compiler/luci/pass/src/SubstitutePadV2ToPadPass.cpp

```diff
@@ -0,0 +1,358 @@
+
+  if (auto transpose = dynamic_cast<luci::CircleTranspose *>(successor))
+  {
+    paddings.apply_transpose(loco::must_cast<luci::CircleConst *>(transpose->perm()));
+    return used_by_maxpool_only(transpose, paddings);
+  }
+  // Support more 'reshaping op' later
+
+  return false;
+}
+
+// Check considion C3)
+bool used_by_maxpool_only(luci::CirclePadV2 *node)
+{
+  Paddings paddings(loco::must_cast<luci::CircleConst *>(node->paddings()));
+
+  return used_by_maxpool_only(node, paddings);
+}
+
+loco::Node *build_pad_from(luci::CirclePadV2 *pad_v2)
+{
+  auto copy_shape = [](const luci::CircleNode *src, luci::CircleNode *dest) {
+    auto rank = src->rank();
+    dest->rank(rank);
+
+    for (decltype(rank) axis = 0; axis < rank; axis++)
+      dest->dim(axis) = src->dim(axis);
+  };
+
+  auto g = pad_v2->graph();
+  auto paddings = loco::must_cast<luci::CircleConst *>(pad_v2->paddings());
+
+  auto pad = g->nodes()->create<luci::CirclePad>();
+  {
+    pad->name(pad_v2->name() + "/pad");
+    luci::add_origin(pad, luci::get_origin(pad_v2));
+
+    pad->dtype(pad_v2->dtype());
+    copy_shape(pad_v2, pad);
+
+    pad->input(pad_v2->input());
+    pad->paddings(pad_v2->paddings());
+  }
+
+  return pad;
+}
+
+bool substitute_pad_v2_to_pad(luci::CirclePadV2 *pad_v2)
+{
+  // precondition
+  {
+    assert(pad_v2);
+    assert(pad_v2->input());
+
+    auto paddings = loco::must_cast<luci::CircleConst *>(pad_v2->paddings());
+    auto constant_values = loco::must_cast<luci::CircleConst *>(pad_v2->constant_values());
+
+    (void)paddings;
+    assert(paddings);
+    assert(paddings->dtype() == loco::DataType::S32);
+    assert(constant_values);
+    assert(constant_values->dtype() == pad_v2->dtype());
+
+    if (not positive_or_zero(pad_v2->input()))
+      return false;
+
+    if (has_all_positive_values(constant_values))
+      return false;
+
+    if (not used_by_maxpool_only(pad_v2))
+      return false;
+  }
+
+  auto pad = build_pad_from(pad_v2);
+
+  replace(pad_v2).with(pad);
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * BEFORE
+ *          |
+ *     [CircleNode] [CircleConst] [CircleConst]
+ *          |             |              |
+ *          -------+----------------------
+ *                 |
+ *            [CirclePadV2]
+ *                 |
+ *          [CircleMaxPool2D]
+ *                 |
+ *
+ * AFTER
+ *          |
+ *     [CircleNode]     [CircleConst]    [CircleNode] [CircleConst]
+ *          |             |      |            |              |
+ *          -------+-------      -------------+--------------+
+ *                 |                          |
+ *            [CirclePad]                [CirclePadV2]
+ *                 |
+ *         [CircleMaxPool2D]
+ *                 |
+ */
+bool SubstitutePadV2ToPadPass::run(loco::Graph *g)
+{
+  bool changed = false;
+  for (auto node : loco::active_nodes(loco::output_nodes(g)))
+  {
+    if (auto circle_node = dynamic_cast<luci::CirclePadV2 *>(node))
```

Comment:
If the network condition is like `PadV2 --> MaxPool2D` then it would be better to check from `CircleMaxPool2D`

Reply:
OK. I will fix.